File: advent_of_code/2022/day_01/solution.py
from utils.py_utils.constants import INPUT
from utils.py_utils.decorators import strip_newlines, to_ints, group_elements, sum_elements, to_list


@to_list()
@sum_elements()
@group_elements()
@to_ints()
@strip_newlines()
def get_data(variation):
    with open(f"inputs/01_{variation}.txt") as f:
        data = f.readlines()
    return data


def easy(data):
    return max(data)


def hard(data):
    return sum(sorted(data)[-3:])


if __name__ == "__main__":
    data = get_data(INPUT)
    print(easy(data))
    print(hard(data))

File: advent_of_code/2022/day_01/solution.rb
require_relative "../utils/ruby_utils/constants"


def raw_data(variation)
  File.readlines("inputs/01_#{variation}.txt",  chomp: true)
end

def modify(data)
  data.slice_when { |element| element == "" }.to_a
      .map { |sub_list| sub_list.map(&:to_i) }
      .map(&:sum)
end

def easy(data)
  modify(data).max
end

def hard(data)
  modify(data).sort[-3..-1].sum
end

if __FILE__ == $0
  data = raw_data(INPUT)
  puts easy(data)
  puts hard(data)
end
